### PR TITLE
feat(story): create mixed content gallery component

### LINKED ISSUE #1337: Extend story gallery to allow mixed content
Story Gallery should support following contents:
- Image
- Video
- Globe
- Embedded (iframe)

### PR DESCRIPTION
**Pull request by andreashelms**
http://localhost:8080/#/stories/debug/7 shows datawrapper chart within iframe

closes #1337

### CODE REVIEW

File: src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  showLightbox: boolean;
+}
+
+const StoryGalleryItem: FunctionComponent<Props> = ({
+  children,
+  currentIndex,
+  showLightbox
+}) => {
+  const containerWidth = children.length * 100;
+  const itemWidth = 100 / children.length;
```
Comment (Immo-Be) [inline src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx:18]: Could you add a comoment why you are multiplying / dividing by 100?

File: src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    styles.slider,
+    showLightbox && styles.lightboxStoryGallery,
+    children.length > 1 && styles.transition
+  );
+
+  return (
+    <div
+      className={imgClasses}
+      style={{
+        width: `${containerWidth}%`,
```
Comment (Immo-Be) [inline src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx:29]: I would suggest we create custom CSS variables and control the value using `document.documentElement.style.setProperty(
      '--container-width'),` - that why we would have better separation of concerns

File: src/scripts/components/stories/story-gallery/story-gallery.tsx [hunk 3]
```diff
@@ -100,11 +91,13 @@ const StoryGallery: FunctionComponent<Props> = ({
...
-        currentIndex={currentIndex}
-        showLightbox={showLightbox}
-      />
+      {children.length > 1 && (
+        <StoryProgress
+          children={children}
+          currentIndex={currentIndex}
+          showLightbox={showLightbox}
+        />
+      )}
```
Comment (Immo-Be) [inline src/scripts/components/stories/story-gallery/story-gallery.tsx:100]:
```suggestion
      {children.length > 1 && (
        <StoryProgress
          currentIndex={currentIndex}
          showLightbox={showLightbox}
        >{children}</>
      )}
```

File: src/scripts/components/stories/story-image/story-image.module.styl [hunk 4]
```diff
@@ -0,0 +1,4 @@
+.photo
+  flex: 1
+  width: 100%
+  height: 0
```
Comment (Immo-Be) [inline src/scripts/components/stories/story-image/story-image.module.styl:4]: `height: 0` ? Could you explain this pattern?

File: src/scripts/components/stories/story/story.tsx [hunk 5]
```diff
@@ -76,49 +74,44 @@ const Story: FunctionComponent = () => {
...
+              case GalleryItemType.Globe:
+                return <StoryGlobe globeItem={item} />;
+              case GalleryItemType.Embedded:
+                return (
+                  <iframe
+                    className={styles.embeddedContent}
+                    src={item.embeddedSrc}></iframe>
+                );
+              default:
+                return <></>;
```
Comment (Immo-Be) [inline src/scripts/components/stories/story/story.tsx:107]: How about we render or log some sort of error message?

File: src/scripts/libs/convert-legacy-story.ts [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+    return [
+      {
+        type: GalleryItemType.Embedded,
+        embeddedSrc: slide.embeddedSrc
+      }
+    ] as EmbeddedItem[];
+  }
+
+  return [];
+};
```
Comment (Immo-Be) [inline src/scripts/libs/convert-legacy-story.ts:60]:
```suggestion
const getGalleryItems = (
  slide: LegacySlide
): VideoItem[] | ImageItem[] | GlobeItem[] | EmbeddedItem[] => {
  if (slide.type === SlideType.Video) {
    return [
      {
        type: GalleryItemType.Video,
        videoId: slide.videoId,
        videoSrc: slide.videoSrc,
        videoCaptions: slide.videoCaptions,
        videoPoster: slide.videoPoster
      }
    ];
  }

  if (slide.type === SlideType.Image && slide.images) {
    return slide.images.map((image, index) => ({
      type: GalleryItemType.Image,
      imageCaption: slide.imageCaptions
        ? slide.imageCaptions[index]
        : // eslint-disable-next-line no-undefined
          undefined,
      image,
      // eslint-disable-next-line no-undefined
      imageFit: slide.imageFits ? slide.imageFits[index] : undefined
    }));
  }

  if (slide.type === SlideType.Globe) {
    return [
      {
        type: GalleryItemType.Globe,
        flyTo: slide.flyTo,
        markers: slide.markers,
        layer: slide.layer,
        layerDescription: slide.layerDescription
      }
    ];
  }

  if (slide.type === SlideType.Embedded) {
    return [
      {
        type: GalleryItemType.Embedded,
        embeddedSrc: slide.embeddedSrc
      }
    ];
  }

  console.warn(`Unknown slide type: ${slide.type}`);
  return [];
};
```

File: src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx (hunk 1 above)
Comment (andreashelms) [reply src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx]: I reused that code from image gallery, but yes, may be helpful here.

File: src/scripts/components/stories/story-image/story-image.module.styl (hunk 4 above)
Comment (andreashelms) [reply src/scripts/components/stories/story-image/story-image.module.styl:4]: Actually I just copied css props form image gallery styles. I am not sure why `height` needs to be 0.

File: src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx (hunk 2 above)
Comment (andreashelms) [reply src/scripts/components/stories/story-gallery-item/story-gallery-item.tsx]: As discussed I think we should keep it as it is to stay consistent. Also I think a global css variable is not needed for this case.

**Review by Immo-Be — APPROVED** [review]

**Pull request merged**